### PR TITLE
LA-671 Advance search in WorkGroup

### PR DESCRIPTION
**Pull request by huynguyennovem**
- Ticket: #671

### CODE REVIEW

**Comment by hoangdat** [comment]
@huynguyennovem @dab246 please review

https://user-images.githubusercontent.com/6462404/139826219-30af6e0d-f8e8-4fd2-9c2b-cc8899eb2ecb.mp4

File: data/lib/src/datasource/shared_space_document_datasource.dart [hunk 1]
```diff
@@ -114,5 +114,5 @@ abstract class SharedSpaceDocumentDataSource {
 
   Future<WorkGroupNode> moveWorkgroupNode(MoveWorkGroupNodeRequest moveRequest, SharedSpaceId sourceSharedSpaceId);
 
-  Future<List<WorkGroupNode?>> advanceSearchWorkgroupNode(SharedSpaceId sharedSpaceId, AdvanceSearchRequest searchRequest);
+  Future<List<WorkGroupNode?>> doAdvanceSearch(SharedSpaceId sharedSpaceId, AdvancedSearchRequest searchRequest);
```
Comment (huynguyennovem) [inline data/lib/src/datasource/shared_space_document_datasource.dart:117]: doAdvanceSearch -> doAdvancedSearch

File: data/lib/src/datasource_impl/local_shared_space_document_datasource_impl.dart [hunk 2]
```diff
@@ -189,7 +189,7 @@ class LocalSharedSpaceDocumentDataSourceImpl implements SharedSpaceDocumentDataS
   }
 
   @override
-  Future<List<WorkGroupNode?>> advanceSearchWorkgroupNode(SharedSpaceId sharedSpaceId, AdvanceSearchRequest searchRequest) {
+  Future<List<WorkGroupNode?>> doAdvanceSearch(SharedSpaceId sharedSpaceId, AdvancedSearchRequest searchRequest) {
```
Comment (huynguyennovem) [inline data/lib/src/datasource_impl/local_shared_space_document_datasource_impl.dart:192]: same

File: data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart [hunk 3]
```diff
@@ -375,7 +375,7 @@ class SharedSpaceDocumentDataSourceImpl implements SharedSpaceDocumentDataSource
   }
 
   @override
-  Future<List<WorkGroupNode?>> advanceSearchWorkgroupNode(SharedSpaceId sharedSpaceId, AdvanceSearchRequest searchRequest) {
+  Future<List<WorkGroupNode?>> doAdvanceSearch(SharedSpaceId sharedSpaceId, AdvancedSearchRequest searchRequest) {
```
Comment (huynguyennovem) [inline data/lib/src/datasource_impl/shared_space_document_datasource_impl.dart:378]: same

File: data/lib/src/network/linshare_http_client.dart [hunk 4]
```diff
@@ -662,7 +662,7 @@ class LinShareHttpClient {
     return SharedSpaceNodeNestedResponse.fromJson(resultJson);
   }
 
-  Future<List<WorkGroupNodeDto>> advanceSearchWorkGroupNodes(SharedSpaceId sharedSpaceId, AdvanceSearchRequest searchRequest) async {
+  Future<List<WorkGroupNodeDto>> advanceSearchWorkGroupNodes(SharedSpaceId sharedSpaceId, AdvancedSearchRequest searchRequest) async {
```
Comment (huynguyennovem) [inline data/lib/src/network/linshare_http_client.dart:665]: advanceSearchWorkGroupNodes -> advancedSearchWorkGroupNodes

File: data/lib/src/repository/sharedspacedocument/shared_space_document_repository_impl.dart [hunk 5]
```diff
@@ -131,8 +131,8 @@ class SharedSpaceDocumentRepositoryImpl implements SharedSpaceDocumentRepository
   }
 
   @override
-  Future<List<WorkGroupNode?>> advanceSearchWorkgroupNode(SharedSpaceId sharedSpaceId, AdvanceSearchRequest searchRequest) {
-    return _sharedSpaceDocumentDataSources[DataSourceType.network]!.advanceSearchWorkgroupNode(sharedSpaceId, searchRequest);
+  Future<List<WorkGroupNode?>> doAdvanceSearch(SharedSpaceId sharedSpaceId, AdvancedSearchRequest searchRequest) {
```
Comment (huynguyennovem) [inline data/lib/src/repository/sharedspacedocument/shared_space_document_repository_impl.dart:134]: same

**Pull request merged**